### ISSUE TITLE
[pickers] Date/DateTime/Time Picker Functionality Issues Post v5->v6 Upgrade

### ISSUE THREAD

**Issue by conerye** [mui/mui-x#10253]
### Order ID

331993142

### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### The problem in depth 🔍

I am having a couple of issues to work through with the Pickers after upgrading from v5 to v6.

1. The first issue is with form submission. Previously, if the TextField was set with the required prop and the user did not enter a value, clicking the submit button set the focus back on the field with the popup, like in this video:

https://github.com/mui/mui-x/assets/39837891/6605e625-2bd6-4dc3-ab11-96d5549c1194

This does not work anymore with v6 pickers... Here is a code sandbox showing this issue:
https://codesandbox.io/s/silent-silence-prhtpt?file=/Demo.js
When submit is clicked, the focus is put back on the field but there is no popup.

2. The second issue I have not been able to recreate (I tried but could not get it in the codesandbox above)... In this scenario, the DatePicker is being used to enter a month/year with minDate Sept 2023, maxDate August 2024 and the date to be entered is August 2024. In order to do that, the user must change the year first but the year view popup 'moves' when the 2024 value is clicked, so the selection does not take/the onChange is not called. If I press and hold the mouse button down and slide the mouse down to 'chase' the 2024 to its new location, then it works. My description is probably not great, so I've captured the behavior in this video, where first I click 2024 and then next try, I press and follow the 2024 with the mouse:

https://github.com/mui/mui-x/assets/39837891/680e4bb7-87dc-4606-a70b-726a931c51ed

3. The last question is on the TimePicker scrollbars
    a. when the scrollbars appear, they are almost on top of the numbers and AM|PM options, any way to add more space between the options and the scrollbar?
    b. there is a bunch of extra space to scroll down to the end of the scrollbar. it looks odd - can this be removed??? 

### Your environment 🌎

<details>
  <summary>`npx @mui/envinfo`</summary>

```
  System:
    OS: Windows 10 10.0.19044
  Binaries:
    Node: 16.15.0 - C:\Program Files\nodejs\node.EXE
    Yarn: Not Found
    npm: 8.5.5 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: Not Found
    Edge: Spartan (44.19041.1266.0), Chromium (116.0.1938.69)
  npmPackages:
    @emotion/react: ^11.11.1 => 11.11.1
    @emotion/styled: ^11.11.0 => 11.11.0
    @mui/base:  5.0.0-beta.12
    @mui/core-downloads-tracker:  5.14.7
    @mui/icons-material: ^5.14.6 => 5.14.6
    @mui/material: ^5.14.6 => 5.14.6
    @mui/private-theming:  5.14.6
    @mui/styled-engine:  5.14.6
    @mui/system: ^5.14.6 => 5.14.6
    @mui/types:  7.2.4
    @mui/utils:  5.14.6
    @mui/x-codemod: ^6.9.0 => 6.9.0
    @mui/x-data-grid:  6.12.0
    @mui/x-data-grid-premium: ^6.12.0 => 6.12.0
    @mui/x-data-grid-pro:  6.12.0
    @mui/x-date-pickers:  6.12.0
    @mui/x-date-pickers-pro: ^6.12.0 => 6.12.0
    @mui/x-license-pro: ^6.10.2 => 6.10.2
    @types/react:  18.0.14
    react: ^18.1.0 => 18.2.0
    react-dom: ^18.1.0 => 18.2.0
    typescript:  4.7.4
```
</details>

**Comment by LukasTy** [comment]
Thank you for creating this issue and using the package! 🙏 
1. The problem is that in v6 we have adopted the `fields` for keyboard date manipulation and that behavior requires a value that we can manipulate, this essentially breaks the native validation behavior.
Do you specifically need the native HTML validation? Or could you get away with implementing custom validation? We have an open issue (https://github.com/mui/mui-x/issues/7633) to introduce the top-level `required` validation attribute to pickers, which should technically cover your use case.
Would such behavior suffice (you'd get an `onError` call with a `required` error message in such case and could provide a desired validation message in your desired way)?
_However, competitors seem to be [handling this case](https://www.telerik.com/kendo-react-ui/components/dateinputs/datepicker/forms/) somehow by still providing the native validation behavior. It means, that we could improve upon our solution._
2. We are aware of this issue and are thinking of ways to solve it. In the mean time, we have provided options on how to avoid the problem [here](https://mui.com/x/react-date-pickers/custom-components/#popper). A solution from our side could actually end up being provided only when we do the Material Design 3 overhaul and provide a different way of choosing the year and month. Could you use the mentioned workaround in the mean time?
3. We also already have an issue (https://github.com/mui/mui-x/issues/9311) to improve on this behavior.
  a. Would applying some custom styles to increase the width of the sections on your end work? 🤔 You could use `styleOverrides` in `themeAugmentation` or an `sx` prop.
  b. That is our design decision to provide an experience, where the selected option is always on the top of the options list. If you do not like this behavior, feel free to[ override it](https://codesandbox.io/s/old-firefly-nk96wt?file=/Demo.tsx). 😉

**Comment by conerye** [comment]
1. We are using the native HTML validation for required fields all over our app, so obvi would not want to go changing all of that now and implementing custom validation just because the Date/DateTime/Time Pickers broke that behavior! I read the open issue you referenced, and yes, adding the required prop to the pickers would be good, but correct me if I am wrong... that would not fix the broken behavior; it would just provide a different way of handling it? this would result in the handling of empty required inputs looking inconsistent across our app...
2. I will try the suggested workaround and get back to you...
3. a. k, when do you think that issue might be resolved?
    b. ok
THANKS

**Comment by LukasTy** [comment]
> 1. <...> but correct me if I am wrong... that would not fix the broken behavior; it would just provide a different way of handling it? this would result in the handling of empty required inputs looking inconsistent across our app...

Yes, you are correct. If we just resolve the issue as described, it won't cover your desired use case of having native `required` validation. For that, we need to look into how to support the behavior in the same way that Telerik does it.

> 3. a. k, when do you think that issue might be resolved?

I'm not sure we will have the capacity to improve on it before the next major. That would put us optimistically in 2024 Q1.
But if see enough traction, maybe we could spare to direct some design resources into revamping it sooner. 🤞

**Comment by conerye** [comment]
ok, so here is where i am at...
1. following open issue (https://github.com/mui/mui-x/issues/7633) - is there a timeframe for resolution?
2. all set but would like to follow - pls provide link for the open issue on this
3 a. following open issue (https://github.com/mui/mui-x/issues/9311) - possible resolution 2024 Q1
   b. k, thanks for the link for overriding... not worth my time to do, we have bigger issues to tackle, lol!

**Comment by LukasTy** [comment]
> 1. <...> is there a timeframe for resolution?

We do not have a concrete timeframe for it as it is a lower-priority change. If possible, it would be safer to delay this change until the next major where we plan a refactor of the field components, which might cause more headache adjusting this behavior to rather than simply implementing it after we complete the refactor. 🤔 🙈 

> 2. pls provide link for the open issue on this

This is probably the best issue to track for such information: https://github.com/mui/mui-x/issues/5490

**Comment by michelengelen** [comment]
This thread bundled three separate reports, and two are now resolved: the year-view "moving target" bug when clicking a value near a viewport edge was fixed by setting a fixed `minHeight` on the calendar (#5490), and the time-picker scrollbar spacing issue was just fixed (#9311, closed this week). The remaining item, native HTML `required` validation on fields, is still open and tracked in #7633; the fields architecture change discussed there hasn't landed yet.

Closing this bundled thread and pointing to the three focused issues above so each stays trackable on its own. Please follow #7633 for the required-validation piece specifically.
